src/tree_builder/mod.rs

```diff
@@ -435,3 +450,157 @@ impl<Handle, Sink> TokenSink
+            self.line_vec.push((name.clone(), self.rcdom.current_line));
+            self.rcdom.create_element(name, attrs)
+        }
+
+        fn create_comment(&mut self, text: StrTendril) -> Handle {
+            self.rcdom.create_comment(text)
+        }
+
+        fn append(&mut self, parent: Handle, child: NodeOrText<Handle>) {
+            self.rcdom.append(parent, child)
+        }
+
+        fn append_before_sibling(&mut self,
+                sibling: Handle,
+                child: NodeOrText<Handle>) -> Result<(), NodeOrText<Handle>> {
+            self.rcdom.append_before_sibling(sibling, child)
+        }
+
+        fn append_doctype_to_document(&mut self,
+                                      name: StrTendril,
+                                      public_id: StrTendril,
+                                      system_id: StrTendril) {
+            self.rcdom.append_doctype_to_document(name, public_id, system_id);
+        }
+
+        fn add_attrs_if_missing(&mut self, target: Handle, attrs: Vec<Attribute>) {
+            self.rcdom.add_attrs_if_missing(target, attrs);
+        }
+
+        fn remove_from_parent(&mut self, target: Handle) {
+            self.rcdom.remove_from_parent(target);
+        }
+
+        fn reparent_children(&mut self, node: Handle, new_parent: Handle) {
+            self.rcdom.reparent_children(node, new_parent);
+        }
+
+        fn mark_script_already_started(&mut self, target: Handle) {
+            self.rcdom.mark_script_already_started(target);
+        }
+
+        fn is_mathml_annotation_xml_integration_point(&self, handle: Self::Handle) -> bool {
+            self.rcdom.is_mathml_annotation_xml_integration_point(handle)
+        }
+
+        fn set_current_line(&mut self, line_number: u64) {
+            self.set_current_line(line_number);
+            self.rcdom.set_current_line(line_number);
+        }
+    }
+
+    impl Default for LineCountingDOM {
+        fn default() -> LineCountingDOM {
+            LineCountingDOM {
+                line_vec: vec!(),
+                rcdom: RcDom::default(),
+            }
+        }
+    }
+
+    #[test]
+    fn check_four_lines() {
+        // Input
+        let sink = LineCountingDOM::default();
+        let opts = ParseOpts::default();
+        let mut resultTok = parse_document(sink, opts);
+        resultTok.process(StrTendril::from("<a>"));
+        resultTok.process(StrTendril::from("</a>"));
+        resultTok.process(StrTendril::from("<b>"));
+        resultTok.process(StrTendril::from("</b>"));
+        // Actual Output
+        let actual = resultTok.finish();
+        // Expected Output
+        let expected = vec![(qualname!(html, "html"), 1),
+                            (qualname!(html, "head"), 1),
+                            (qualname!(html, "body"), 1),
+                            (qualname!(html, "a"), 1),
+                            (qualname!(html, "b"), 1)];
```

Comment:
Our expected output will want to ensure that at least some elements are associated with different lines.